tests/ui/match/ref_pat_eat_one_layer_2024/ref_pat_eat_one_layer_2021.rs

```diff
@@ -0,0 +1,15 @@
+//@ run-pass
+//@ edition: 2021
+//@ revisions: classic structural both
+#![allow(incomplete_features)]
+#![cfg_attr(any(classic, both), feature(ref_pat_eat_one_layer_2024))]
+#![cfg_attr(any(structural, both), feature(ref_pat_eat_one_layer_2024_structural))]
+
+pub fn main() {
+    if let &Some(Some(x)) = &Some(&mut Some(0)) {
+        let _: &u32 = x;
```

Comment:
This test would pass even if we ended up with a `&mut u32` here, e.g.:

```rust
let x = &mut ();
let _: &() = x; //~ OK
```

Reply:
It would fail borrowck as `&mut` is not `Copy`

Reply:
But note that:

```rust
    let x = &mut ();
    let _: &&() = &x; //~ ERROR mismatched types
```

So perhaps we could write:

```suggestion
        let _: &&u32 = &x;
```

Reply:
Makes sense.